### PR TITLE
chore!: bump to rsmt2d v0.10.0

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/2060

Marked as breaking b/c after this PR `NewDataAvailabilityHeader` may return an error.

### CODE REVIEW

File: app/process_proposal.go [hunk 1]
```diff
@@ -122,7 +122,11 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) (resp abci.Resp
 		return reject()
 	}
 
-	dah := da.NewDataAvailabilityHeader(eds)
+	dah, err := da.NewDataAvailabilityHeader(eds)
```
Comment (staheri14) [inline app/process_proposal.go:125]: [Question] I'm curious to know if a malicious player can cause a validator to panic by using corrupted data. If that is indeed possible, would triggering a panic be considered a reasonable response?

**Review by staheri14 — APPROVED** [review]
I left a question about the panic behavior, it is non-blocking if we are sure that panic is a safe option.

File: app/process_proposal.go (hunk 1 above)
Comment (rootulp) [reply app/process_proposal.go:125]: I don't understand the second part of the question, do you mind rephrasing?

I don't think it's possible for a malicious proposer to cause a validator to panic. If it were possible, then a malicious proposer could halt the chain by causing all honest validators to `panic`. I don't think `panic` is a reasonable response for an honest validator if they encounter a malicious block. In this scenario, an honest validator should reject the block and remain online.

Comment (staheri14) [reply app/process_proposal.go:125]: > If that is indeed possible, would triggering a panic be considered a reasonable response?

Thanks for the clarification. 🙏 
My question about panic does not apply to `process_proposal` as unlike `prepare_proposal` it does not panic (confused the two).

**Pull request merged**